controller/src/beerocks/master/tasks/dynamic_channel_selection_task.h

```diff
@@ -29,34 +29,28 @@ class dynamic_channel_selection_task : public task, public std::enable_shared_fr
                                    task_pool &tasks_, std::string radio_mac_);
     virtual ~dynamic_channel_selection_task() {}
 
-    typedef struct {
+    struct sTriggerSingleScan_event {
         sMacAddr radio_mac;
-    } sTriggerSingleScan_event;
-
-    typedef struct {
+    };
+    struct sScanTriggered_event {
```

Comment:
I think it would be much better to define a single event struct and an enum for the event type, instead of duplicating event structures like that.

```
enum class eDcsEventType { SCAN_TRIGGERED, SINGLE_SCAN, ...};

struct sDcsEvent {
    eDcsEventType type;
    sMacAddr radio_mac;
    union data {
        sChannelScanResultsElement scan_results;
        ...
    };
```

Reply:
done